### ISSUE TITLE
Fedora 32 ICG cannot find clang libs during trick build

### ISSUE THREAD

**Issue by spfennell** [nasa/trick#989]
```
/usr/bin/g++ -o /__w/trick/trick/bin/trick-ICG object_Linux_10_x86_64/PrintFileContents10.o object_Linux_10_x86_64/ICGASTConsumer.o object_Linux_10_x86_64/Utilities.o object_Linux_10_x86_64/CommentSaver.o object_Linux_10_x86_64/TypedefVisitor.o object_Linux_10_x86_64/EnumVisitor.o object_Linux_10_x86_64/HeaderSearchDirs.o object_Linux_10_x86_64/ClassVisitor.o object_Linux_10_x86_64/ClassValues.o object_Linux_10_x86_64/FieldVisitor.o object_Linux_10_x86_64/PrintAttributes.o object_Linux_10_x86_64/FindTrickICG.o object_Linux_10_x86_64/ConstructValues.o object_Linux_10_x86_64/EnumValues.o object_Linux_10_x86_64/TranslationUnitVisitor.o object_Linux_10_x86_64/main.o object_Linux_10_x86_64/PrintFileContentsBase.o object_Linux_10_x86_64/VariableVisitor.o object_Linux_10_x86_64/FieldDescription.o object_Linux_10_x86_64/ClassTemplateVisitor.o object_Linux_10_x86_64/map_trick_units_to_udunits.o -lclangFrontend -lclangDriver -lclangSerialization -lclangParse -lclangSema -lclangAnalysis -lclangEdit -lclangAST -lclangLex -lclangBasic  -lLLVM-10  -L/usr/lib64  -ludunits2
/usr/bin/ld: cannot find -lclangFrontend
/usr/bin/ld: cannot find -lclangDriver
/usr/bin/ld: cannot find -lclangSerialization
/usr/bin/ld: cannot find -lclangParse
/usr/bin/ld: cannot find -lclangSema
/usr/bin/ld: cannot find -lclangAnalysis
/usr/bin/ld: cannot find -lclangEdit
/usr/bin/ld: cannot find -lclangAST
/usr/bin/ld: cannot find -lclangLex
/usr/bin/ld: cannot find -lclangBasic
collect2: error: ld returned 1 exit status
```

**Comment by spfennell** [comment]
clang-devel on Fedora 32 comes from the updates-testing repo instead of the updates repo, so this is probably not our fault. Removing Fedora 32 test for now until it is a stable release.

**Comment by spfennell** [comment]
Fedora 31 (scroll to the right to see repo):
```
 Package                                   Arch    Version                                     Repository        Size
=====================================================================================================================
Installing:
 bison                                     x86_64  3.4.1-2.fc31                                fedora           779 k
 clang                                     x86_64  9.0.1-2.fc31                                updates          379 k
 clang-devel                               x86_64  9.0.1-2.fc31                                updates          1.6 M
...
```
Fedora 32:
```
============================================================================================================
 Package                                   Arch    Version                            Repository        Size
============================================================================================================
Installing:
 bison                                     x86_64  3.5-2.fc32                         fedora           818 k
 clang                                     x86_64  10.0.0-1.fc32                      updates-testing   86 k
 clang-devel                               x86_64  10.0.0-1.fc32                      updates-testing  1.7 M
...
```

**Comment by ABION-Technology** [comment]
We are currently using Trick at my employment, we have a RHEL network where we do our primary software development, and also recently we setup Virtual Machines with Red Hat CentOS 6 and VirtualBox so we can still do software development when the RHEL network is down for maintenance or upgrades. We are using Trick 15 (2015 version) which is still available on this Github, even though it is not the most recent version . Trick comes with a configure script which has to run before you can do a "make" with it in the root directory. I encountered a very unusual error in the configure script where it could not find clang libraries in the directory /usr/include/clang, the directory is actually non-existent even after installing clang, clang-devel, and llvm, so I had a few questions since this is similar to this issue with Fedora Linux.

1) I noticed the zip file for Trick 15 already came with binaries in a bin folder, so does this mean that Trick can be used right "out of the box"? and a "make" in the root folder is not needed? Or does a user still have to do a "make" in the Trick root folder to build Trick?

2) The configure script gave me several errors for applications not found on the box and I was able to correct that by installing the needed applications, like Flex and Bison, but also there was a warning about Google Test not being found, only a warning, not an error, will Trick still run without having Google Test installed?

3) Since Red Hat CentOS and RHEL are basically the same a coworker has suggested that we just copy Trick from our Red Hat RHEL network to our Red Hat CentOS VM and he assumes it will work after this simple copy operation, I also had a question about this, do we simply copy the binaries in the bin folder? or do we also have to copy over all the folders as well?

4) some basic research I did seems to point to the fact that there is never a /usr/include/clang folder and instead clang uses libraries from other compilers, which makes sense since clang is basically a front end for other compilers like gcc ad g++,  I am just confused why the configure script is generating this error and expecting there to be a /usr/include/clang directory to be present?

Thanks, these questions are kind of similar to this issue with Fedora, that is why I posted them here.

**Comment by Michael-G-Workman** [comment]
Hello, I had a few questions about Trick which are similar to this issue. We are using Trick at my employment where it is currently installed on our Red Hat Enterprise Linux (RHEL) developer network, we are using Trick 15, also, we recently setup Red Hat CentOS 6 Virtual Machines with Virtual Box so we can continue to do software development when our RHEL network is down for maintenance or upgrades. So I downloaded Trick 15 from this GitHub, and began the process of running the included configure script required before doing a "make"

1 - I encountered an error similar to the one for this issue when the configure script is run, and it fails to find a directory /usr/include/clang on the localhost, this folder is nonexistent even after installing clang, clang-devel, and llvm. research says that since clang is a front end for other compilers it uses the include directory for the other compiler rather than its own, so I was just curious to know what am I missing that is generating this error?

2 - Since Red Hat CentOS is identical to RHEL, a coworker has suggested we just copy the already built and installed Trick from our RHEL to our CentOS VM and it should then run correctly. My question is do we just copy over the binaries in the bin directory? or do we also have to copy over all the other directories and their contents?

3 - I noticed in the downloaded zip file from this GitHub that there is binaries in the bin folder already, does this mean that Trick is ready to run "out of the box" without running the configure script and doing a "make"? or does a user still have to run the configure script and do a "make"?

4 - The configure script generated some errors when it did not find applications needed for Trick, I was able to work through those errors by installing the missing applications, like Flex and Bison. but the script also generated a WARNING, not an ERROR, when it did not find Google Test, since this is just a warning does that mean Trick will work correctly without Google Test? or not?

Thanks, these questions are similar to this issue so that is why I did not post them in a new issue.

**Comment by spfennell** [comment]
1. I am not sure why you are getting this error. I haven't worked a lot with Trick 15, maybe @alexlin0, @dbankieris, or @jmpenn could chime in on this. CentOS 6 is nearly EOL, so some of the packages they used to provide are starting to disappear because of dependency issues.

I do not believe this has to do with your clang error, but I know that we required devtoolset-3-gcc-c++ to be installed, which is a part of Software Collections and hard to find these days (Scientific Linux no longer provides it). Maybe you could try the version in this repo: https://copr.fedorainfracloud.org/coprs/rhscl/devtoolset-3/

2. This might work, however you will still need most of the Trick dependencies to compile sims. You should copy the entire trick directory, there are some header files and libraries that are required for building sims.

3. The files in the bin folder already are perl or bash scripts that do not need to be compiled. Most of the important ones will not work at all until you have "./configure" and "make".

4. gtest is optional and not required for trick.

#### Here are the old old install instructions for Trick 15. Already I can tell you that the instructions for devtoolset-3-gcc-c++ will no longer work.

Let me know if there is anything else I can do to help.

#### RedHat Enterprise Linux (RHEL) 6

Trick requires gcc version 4.8+ to compile.  gcc 4.9 is available through RHEL's [Software Collections](https://access.redhat.com/documentation/en/red-hat-software-collections/) capability.  To install the software collections, the software collections reposotory needs to be added as a yum repository and the scl-utils package needs to be optionally installed.  The instructions below are specific to Scientific Linux 6, an RHEL derivative.  Repository location will be different for the official RHEL 6 and CentOS.

```bash
wget http://ftp.scientificlinux.org/linux/scientific/6x/external_products/softwarecollections/yum-conf-softwarecollections-1.0-1.el6.noarch.rpm
rpm -ivh yum-conf-softwarecollections-1.0-1.el6.noarch.rpm
# From the software collections repository
yum install devtoolset-3-gcc-c++
```

Trick requires the clang/llvm compiler to compile and link the Trick Interface Code Generator.  clang/llvm is available through the [Extra Packages for Enterprise Linux](https://fedoraproject.org/wiki/EPEL) repository.  Download and install the 'epel-release' package.

```bash
# From the EPEL repository
yum install llvm llvm-devel llvm-static clang clang-devel
```

Trick also requires development packages from the base and epel repositories

```bash
# From the base repository
yum install bison flex gcc gcc-c++ libxml2-devel make \
 openmotif openmotif-devel python-devel perl swig zlib-devel
#UDUnits is required and is in the EPEL repository
yum install udunits2 udunits2-devel
```

The javac compiler must be installed.  The javac compiler is included with the Java Development Kit (JDK).  Trick will work with either the Oracle JDK or OpenJDK, but we prefer the Oracle JDK. 
At the time of this writing (March 2015) the Oracle JDK 1.8 is available from the [Oracle download site](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html).

Alternatively OpenJDK is available through yum

```bash
yum install java-1.8.0-openjdk
```

Trick makes use of several optional packages if they are present on the system.  These include using the HDF5 package for logging, the GSL packages for random number generation, and google test (gtest) for Trick's unit testing.  These are available from the EPEL repository

```bash
yum install hdf5-devel gsl-devel gtest-devel
```

**Comment by spfennell** [comment]
Also for 1. You are probably missing the static libraries in llvm-static

**Comment by Michael-G-Workman** [comment]
Ok, thanks for your help, reviewing your input today. :-)

**Comment by Michael-G-Workman** [comment]
So yes, I did see that gcc 4.8 is required for Trick, but is that requirement only for the most recent version of Trick? what about Trick 15? we have GNU compiler collection (GCC) installed on our RHEL development network version 4.4.7, which is from 2012, and the time stamps from our Trick on our RHEL network are 2015, so I am assuming that that the 4.4.7 version of gcc was used to compile it on our RHEL network, so I was interested to know the version of GCC that is required for Trick 15?

**Comment by spfennell** [comment]
I believe 15.0 and 15.1 do not require gcc 4.8. 15.2 and later require GCC 4.8+

**Comment by Michael-G-Workman** [comment]
Ok thanks :-)

**Comment by spfennell** [comment]
will be resolved by #1009 or related issues. Clang linked libraries have changed on some systems.

**Comment by spfennell** [comment]
#1040 has partially addressed  this issue. Unfortunately linking in libclang-cpp has created some other issues with ICG, see #1041
